zkstore/ident.go

```diff
@@ -0,0 +1,50 @@
+package zkstore
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// Ident specifies the location of a stored item
+type Ident struct {
+	// Location points to where an item lives in the store.
+	Location Location
+
+	// Version, if specified, specifies a named version of the data
+	Version string
+
+	// ZKVersion specifies the ZK version of the data.  This will be
+	// used to prevent accidental overwrites.  If ZKVersion is nil,
+	// then the version will not be considered for operations.
+	ZKVersion *int32
```

Comment:
I remember we said we didn't want to be tied to a specific implementation for the library API (here ZK), as such should we rename this field to something more generic? The error being associated to this field is called `ErrVersionConflict`, which doesn't mention ZK.

It might be confusing to have the "version" semantic in 2 Ident fields, what about renaming `ZKVersion` to `ID` or `TransactionID`? And the associated error to `ErrStaleTransaction` or similar.